### PR TITLE
Added Cylc Doc to Install Cylc Components

### PR DESCRIPTION
**Pull request by wxtim**

### CODE REVIEW

File: install-cylc-components/action.yml [hunk 1]
```diff
@@ -99,6 +113,7 @@ runs:
         cylc_uis:  ${{ inputs.cylc_uiserver_tag || steps.get_tag.outputs.cylc_uiserver }}
         meto_rose: ${{ inputs.metomi_rose_tag   || steps.get_tag.outputs.rose }}
         cylc_rose: ${{ inputs.cylc_rose_tag     || steps.get_tag.outputs.cylc_rose }}
+        cylc_rose: ${{ inputs.cylc_doc_tag     || steps.get_tag.outputs.cylc_doc }}
```
Comment (MetRonnie) [inline install-cylc-components/action.yml]:
```suggestion
        cylc_doc: ${{ inputs.cylc_doc_tag     || steps.get_tag.outputs.cylc_doc }}
```

File: install-cylc-components/action.yml [hunk 2]
```diff
@@ -122,6 +137,11 @@ runs:
             "cylc-rose[${{ inputs.cylc_rose_opts }}] @ git+https://github.com/${{ inputs.cylc_rose_repo }}@${cylc_rose}" \
             >> "$REQS"
         fi
+        if [[ ${{ inputs.cylc_rose }} == true ]]; then
```
Comment (MetRonnie) [inline install-cylc-components/action.yml]:
```suggestion
        if [[ ${{ inputs.cylc_doc}} == true ]]; then
```

**Review by MetRonnie — CHANGES_REQUESTED** [review]

**Review by MetRonnie — APPROVED** [review]
Looks good, do you want to test it out in your fork of cylc-doc before merging?

**Comment by wxtim** [comment]
> Looks good, do you want to test it out in your fork of cylc-doc before merging?

Unfortunately it looks a bit more complex that that - 

Checking out the right branch is simple - I used `cylc_doc_tag: ${{github.sha}}`. Then I realized that I needed to at least check out `./docs/.github/actions/install-dependencies` to install the deps Pip cannot handle. Having done that, I got this problem - ~any thoughts ; https://github.com/wxtim/cylc-doc/actions/runs/6417167750/job/17422350311~ My personal gh-pages branch is well and truly stale.

**Comment by MetRonnie** [comment]
> Checking out the right branch is simple - I used `cylc_doc_tag: ${{github.sha}}`

I don't think that's right. The point of this is you don't specify the `cylc_doc_tag` - it is automatically selected

**Comment by wxtim** [comment]
> > Checking out the right branch is simple - I used `cylc_doc_tag: ${{github.sha}}`
> 
> I don't think that's right. The point of this is you don't specify the `cylc_doc_tag` - it is automatically selected

Won't it automatically specify the latest version, not the branch under test

**Comment by wxtim** [comment]
> > Checking out the right branch is simple - I used `cylc_doc_tag: ${{github.sha}}`
> 
> I don't think that's right. The point of this is you don't specify the `cylc_doc_tag` - it is automatically selected

Agreed - my mistake

I can't get mine to go past the git rebase step, but I think that the critical parts work https://github.com/wxtim/cylc-doc/actions/runs/6418414801/job/17426157554

**Comment by MetRonnie** [comment]
That could be because your `gh-pages` branch on your fork is quite far behind the main repo?

**Comment by wxtim** [comment]
> That could be because your `gh-pages` branch on your fork is quite far behind the main repo?

I thought that I'd synced it

**Comment by MetRonnie** [comment]
image

**Comment by wxtim** [comment]
@MetRonnie  - I think GH has decided that I'm bitcoin mining - none of my workflows are running!

**Comment by MetRonnie** [comment]
What's the status with this? Did you successfully test it out?

**Comment by wxtim** [comment]
> What's the status with this? Did you successfully test it out?

No.  I'm not at all sure that my branches are right for that. 

I'm also not sure that the nightly build isn't mostly working now.

**Comment by MetRonnie** [comment]
> I'm also not sure that the nightly build isn't mostly working now.

Come again?

Even if nightly is passing, we know it's always building from master even when it's supposed to be 8.2

**Comment by wxtim** [comment]
> > I'm also not sure that the nightly build isn't mostly working now.
> 
> Come again?
> 
> Even if nightly is passing, we know it's always building from master even when it's supposed to be 8.

Ah. Forgot that bit.

**Comment by wxtim** [comment]
Looks like it's partially working on https://github.com/wxtim/cylc-doc/actions/runs/6611755158/job/17956303415 - although I'm not sure about the error on 8.2.x branches - I've reproduced something like this locally.

**Comment by MetRonnie** [comment]
> Looks like it's partially working on https://github.com/wxtim/cylc-doc/actions/runs/6611755158/job/17956303415 - although I'm not sure about the error on 8.2.x branches - I've reproduced something like this locally.

`ERROR: An option to autoclass is either unknown or has an invalid value: 'no-index'`

Your cylc-doc nightly workflow is still checking out master - it needs to check out the branch as determined by this action

**Comment by wxtim** [comment]
> Your cylc-doc nightly workflow is still checking out master - it needs to check out the branch as determined by this action

The trouble here is that we need to run `cylc-doc:.github/actions/install-dependencies` to allow the Cylc component to install properly. I think that I've successfully bodged a demo in https://github.com/wxtim/cylc-doc/actions/runs/6640353536/job/18040602680 which shows this change working correctly, but I'm not sure if we can use it without moving `install-dependencies` (it's Pygraphviz again) to another repo?

**Comment by MetRonnie** [comment]
Can we not just run `actions/checkout` twice: once at the beginning to checkout master, then once after this action has run to checkout the correct branch?

**Comment by wxtim** [comment]
Finally https://github.com/wxtim/cylc-doc/actions/runs/6664794951/job/18113098974

I think that the 8.2-jps-V1 is broken for other reasons.

**Pull request closed without merging**